plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -234,6 +253,129 @@ public void onFailure(Exception e) {
         });
     }
 
+    private void checkAgentBeforeDeleteModel(String modelId, ActionListener<Boolean> actionListener) {
```

Comment:
We don't want to expose hidden agent ids to the user. Have you tested? What's your opinion @ylwu-amzn @dhrubo-os  @zane-neo ?

Reply:
As it's hidden it could be an issue to expose the id, better filter out this.

Reply:
Done.